optimism/FunctionSpace.py

```diff
@@ -193,7 +193,14 @@ def interpolate_to_points(functionSpace, nodalField):
     return jax.vmap(interpolate_to_element_points, (None, 0, 0))(nodalField, functionSpace.shapes, functionSpace.mesh.conns)
 
 
-def integrate_over_block(functionSpace, U, stateVars, dt, func, block,
+def vmapPropValue(propArray):
+    numAxes = len(propArray.shape)
+    if numAxes > 1:
+        return 0
+    else:
+        return None
+
+def integrate_over_block(functionSpace, U, stateVars, props, dt, func, block,
                          *params, modify_element_gradient=default_modify_element_gradient):
```

Comment:
@ralberd If we also append X (the nodal coordinates) to methods like these, you won't need a special adjoint system. But let's save that for later.